### PR TITLE
Document Windows NPM setup and make sure it starts

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

- Add documentation about the need to use `windows_npm_install` when setting up NPM on Windows.
- Restart Datadog Agent service on Windows on system probe configuration changes.

### Motivation

- Flag was undocumented and we did not mention this explicitly anywhere.
- The NPM driver did not start since the system probe class was applied after the service start.

### Additional Notes

It's difficult to do this automatically since we would have to call the `datadog::datadog_agent` class on the `datadog::system_probe` one, so instead we are just documenting this.

### Describe your test plan

Docs only change.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -281,6 +294,7 @@ These variables can be set in the `datadog_agent` class to control settings in t
 | `scrub_args`                            | A boolean to enable the process cmdline scrubbing (defaults to true).                                                                                                                            |
 | `custom_sensitive_words`                | An array to add more words beyond the default ones used by the scrubbing feature (defaults to `[]`).                                                                                             |
 | `logs_enabled`                          | A boolean to enable the logs Agent (defaults to false).                                                                                                                                          |
+| `windows_npm_install`                   | A boolean to enable the Windows NPM driver installation (defaults to false)                                                                                                                      |
```
Comment (buraizu) [inline README.md:297]:
```suggestion
| `windows_npm_install`                   | A boolean to enable the Windows NPM driver installation (defaults to false).                                                                                                                      |
```

**Review by buraizu — APPROVED** [review]
Approving, but left a small suggestion for consistency with other parameter descriptions

File: README.md [hunk 2]
```diff
@@ -194,6 +194,20 @@ With the [`ini_setting` module](https://forge.puppet.com/modules/puppetlabs/inif
 
 5. Verify your Puppet data is in Datadog by searching for `sources:puppet` in the [Event Stream][5].
 
+### NPM setup
+
+To enable the Datadog Agent Network Performance Monitoring (NPM) features follow these steps:
+
+1. (Windows only) If the Agent is already installed, uninstall it by passing `win_ensure => absent` to the main class.
```
Comment (KSerrania) [inline README.md:201]: This fails if the `datadog_agent::system_probe` class is defined, with:
```
Error: Could not find resource 'File[C:/ProgramData/Datadog]' in parameter 'require' (file: C:/ProgramData/PuppetLabs/code/environments/production/modules/datadog_agent/manifests/system_probe.pp, line: 35) on node <node name>
```
so we should also mention that you also need to remove the `datadog_agent::system_probe` class.

File: README.md [hunk 3]
```diff
@@ -194,6 +194,20 @@ With the [`ini_setting` module](https://forge.puppet.com/modules/puppetlabs/inif
...
+2. (Windows only) Pass the `windows_npm_install` option with value `true` to the `datadog::datadog_agent` class. Remove `win_ensure` if added on previous step.
+3. Use the `datadog_agent::system_probe` class to properly create the configuration file:
+
+```conf
+class { 'datadog_agent::system_probe':
+    enabled => true, 
+}
```
Comment (KSerrania) [inline README.md:208]: It seems like, on Windows, at least with recent versions, `enabled => true` doesn't make the system-probe component start, and you need:
```
class { 'datadog_agent::system_probe':
    network_enabled => true, 
}
```
instead.

This is probably due to the fact that not setting `network_enabled => true` makes puppet write:
```
network_config:
  enabled: false
```
explicitly in the config.

**Review by KSerrania — COMMENTED** [review]
I've left some comments about the documentation, but otherwise this looks great!

File: README.md (hunk 2 above)
Comment (mx-psi) [reply README.md:201]:
```suggestion
1. (Windows only) If the Agent is already installed, uninstall it by passing `win_ensure => absent` to the main class and removing other classes.
```

File: README.md (hunk 3 above)
Comment (mx-psi) [inline README.md:208]:
```suggestion
class { 'datadog_agent::system_probe':
    network_enabled => true, 
}
```

**Review by KSerrania — APPROVED** [review]

**Pull request merged**